src/hooks/use-snapshots.tsx

```diff
@@ -0,0 +1,376 @@
+	) => Promise< void >;
+	deleteAllSnapshots: ( snapshots: Pick< Snapshot, 'atomicSiteId' >[] ) => Promise< void >;
+	clearFloatingSnapshots: ( allSnapshots: Pick< Snapshot, 'atomicSiteId' >[] ) => void;
+	isLoading: boolean;
+	loadingDeletingAllSnapshots: boolean;
+	loadingServerSnapshots: boolean;
+	activeSnapshotCount: number;
+	snapshotQuota: number;
+	fetchSnapshotUsage: () => Promise< { site_limit: number; site_count: number } | undefined >;
+	isLoadingSnapshotUsage: boolean;
+	initiated: boolean;
+}
+
+export interface SnapshotStatusResponse {
+	is_deleted: string;
+	domain_name: string;
+	atomic_site_id: string;
+	status: '1' | '2';
+}
+
+interface FetchSnapshotResponse {
+	sites: { atomic_site_id: number }[];
+}
+
+interface WpcomNetworkError extends Error {
+	code?: string;
+}
+
+export const SnapshotContext = createContext< SnapshotContextType >( {
+	snapshots: [],
+	allSnapshots: null,
+	addSnapshot: () => undefined,
+	updateSnapshot: () => undefined,
+	removeSnapshot: () => undefined,
+	fetchAllSnapshots: async () => [],
+	deleteSnapshot: async () => undefined,
+	deleteAllSnapshots: async () => undefined,
+	clearFloatingSnapshots: () => undefined,
+	isLoading: false,
+	loadingDeletingAllSnapshots: false,
+	loadingServerSnapshots: false,
+	activeSnapshotCount: 0,
+	snapshotQuota: LIMIT_OF_ZIP_SITES_PER_USER,
+	fetchSnapshotUsage: async () => undefined,
+	isLoadingSnapshotUsage: false,
+	initiated: false,
+} );
+
+export const SnapshotProvider: React.FC< { children: ReactNode } > = ( { children } ) => {
+	const [ snapshots, setSnapshots ] = useState< Snapshot[] >( [] );
+	const [ initiated, setInitiated ] = useState( false );
+	const [ isLoading, setIsLoading ] = useState( false );
+	const [ loadingDeletingAllSnapshots, setLoadingDeletingAllSnapshots ] = useState( false );
+	const [ loadingServerSnapshots, setLoadingServerSnapshots ] = useState( false );
+	const [ activeSnapshotCount, setActiveSnapshotCount ] = useState( 0 );
+	const [ allSnapshots, setAllSnapshots ] = useState< Pick< Snapshot, 'atomicSiteId' >[] | null >(
+		null
+	);
+	const [ snapshotQuota, setSnapshotQuota ] = useState( LIMIT_OF_ZIP_SITES_PER_USER );
+	const [ isLoadingSnapshotUsage, setIsLoadingSnapshotUsage ] = useState( false );
+	const { client } = useAuth();
+	const isOffline = useOffline();
+	const { __ } = useI18n();
+
+	const removeSnapshot = useCallback( ( snapshot: Pick< Snapshot, 'atomicSiteId' > ) => {
+		setSnapshots( ( prevSnapshots ) =>
+			prevSnapshots.filter( ( s ) => s.atomicSiteId !== snapshot.atomicSiteId )
+		);
+		setActiveSnapshotCount( ( prevCount ) => ( prevCount === 0 ? prevCount : prevCount - 1 ) );
+	}, [] );
+
+	useEffect( () => {
+		if ( ! initiated ) {
+			return;
+		}
+		getIpcApi().saveSnapshotsToStorage( snapshots );
+	}, [ snapshots, initiated ] );
+
+	useLayoutEffect( () => {
+		getIpcApi()
+			.getSnapshots()
+			.then( ( storedSnapshots ) => {
+				setSnapshots( storedSnapshots );
+				setInitiated( true );
+			} );
+	}, [] );
```

Comment:
Out of curiosity,  What's the benefit of using `useLayoutEffect` ?

Reply:
`useEffect` runs on initial render, and `useLayoutEffect` runs before that. As this is just to retrieve data, and better have them available when the UI is getting rendered, I found it to be better.
TL;DR: Get our data faster, before the initial render.